modules/client/src/addressList/internal/client/estimation.ts

```diff
@@ -36,10 +38,28 @@ export class ClientAddressListEstimation extends ClientCore
     } else if (!signer.provider) {
       throw new Error("A web3 provider is needed");
     }
-    // TODO: Implement
-    return Promise.resolve(
-      this.web3.getApproximateGasFee(Random.getBigInt(BigInt(1500))),
+
+    const addresslistContract = AllowlistVoting__factory.connect(
+      _params.pluginAddress,
+      signer,
+    );
+
+    let cid = "";
```

Comment:
In line with the point that you raised, we will definitely need to decouple pinning from creating the DAO :100: 
`APP-1286` to the rescue

```suggestion
    // TODO: Compute the cid instead of pinning the data on the cluster
    let cid = "";
```